woocommerce/woocommerce
# Add react-powered main payments settings screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/50721.


This PR replicates the existing payment methods screen in the new React component (as much as possible).

It provides the following functionalities:

- It pre-fills the WooPayments gateway when store is eligible for it.
- Show the payment methods in a table by fetching the payment methods from the server.
- Can enable/disable payment methods.
- Show additional payment providers in the last row.

It doesn't include the following:

- The re-order functionality.
- The 3-dots menu on the WooPayments row.
- The sub menu items `WooPayments |All payment methods` when WooPayments is enabled.
- The Save button since re-order functionality is not supported



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In a fresh site with beta tester plugin installed, go through onboarding wizard and set country to Afghanistan.
2. Go to `Tools > WCA Tester Helper` and enable the feature flag `reactify-classic-payments-settings`.
2. Go to `WooCommerce > Settings > Payments`
3. Observe the Main payments screen shows the payment methods in a table without WooPayments.
4. Change store country to United States.
5. Observe the Main payments screen shows the payment methods in a table with WooPayments.
6. Enable/disable payment methods and observe the changes.
7. Click the payment method and confirm it redirects to the corresponding settings page.
8. Click on WCPay `Install` button and confirm it redirects to the payment page after installation.
9. Go back to the payments screen and confirm the WooPayments row is updated. 
10. Observe that WooPayments banner (`Payments made simple, designed exclusively for WooCommerce stores.`) is shown

#### Before installing WCPay:

Classic:

![Screenshot 2024-08-21 at 12 41 20](https://github.com/user-attachments/assets/8e5dd04c-bc5a-4ed6-99b7-c5523a36535f)


React:

![Screenshot 2024-08-21 at 12 41 32](https://github.com/user-attachments/assets/fd0ed9f2-9f58-4992-91fc-75311544bc5a)

#### After installing WCPay:


Classic:

![Screenshot 2024-08-21 at 12 41 03](https://github.com/user-attachments/assets/6ad859f4-a54a-45a3-87e7-6d67f10f4ace)

React:
![Screenshot 2024-08-21 at 12 40 46](https://github.com/user-attachments/assets/a635692b-8479-4e31-b2c9-08ab2095a32f)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
